src/main/java/com/avispl/symphony/dal/avdevices/encoderdecoder/haivision/xdecoder/HaivisionXDecoderCommunicator.java

```diff
@@ -692,31 +870,1955 @@ public boolean validatePortRange(Integer portNumber) {
+						createStream.setSourceAddress(DecoderConstant.ADDRESS_ANY);
+						break;
+					case MULTI_CAST:
+						createStream.setDestinationAddress(DecoderConstant.EMPTY);
+						createStream.setSourceAddress(DecoderConstant.EMPTY);
+						break;
+					default:
+						if (logger.isWarnEnabled()) {
+							logger.warn(String.format("SRT mode %s is not supported.", networkType.getUiName()));
+						}
+						break;
+				}
+
+				populateCreateStreamControl(stats, advancedControllableProperties, createStream, streamControllingGroup);
+				break;
+			case PORT:
+				String port = DecoderConstant.EMPTY;
+				try {
+					Integer portIntValue = Integer.parseInt(value);
+					if (portIntValue < DecoderConstant.MIN_PORT) {
+						portIntValue = DecoderConstant.MIN_PORT;
+					}
+					if (portIntValue > DecoderConstant.MAX_PORT) {
+						portIntValue = DecoderConstant.MAX_PORT;
+					}
+					port = portIntValue.toString();
+				} catch (Exception e) {
+					if (logger.isWarnEnabled()) {
+						logger.warn("Invalid port value", e);
+					}
+				}
+				createStream.setPort(port);
+				createStream.setDestinationPort(port);
+
+				addAdvanceControlProperties(advancedControllableProperties,
+						createNumeric(stats, streamControllingGroup + StreamControllingMetric.PORT.getName(), port));
+				populateCancelButtonForCreateStream(stats, advancedControllableProperties);
+				break;
+			case ADDRESS:
+				createStream.setDestinationAddress(value);
+
+				addAdvanceControlProperties(advancedControllableProperties,
+						createText(stats, streamControllingGroup + StreamControllingMetric.ADDRESS.getName(), value));
+				populateCancelButtonForCreateStream(stats, advancedControllableProperties);
+				break;
+			case MULTICAST_ADDRESS:
+				createStream.setDestinationAddress(value);
+
+				addAdvanceControlProperties(advancedControllableProperties,
+						createText(stats, streamControllingGroup + StreamControllingMetric.MULTICAST_ADDRESS.getName(), value));
+				populateCancelButtonForCreateStream(stats, advancedControllableProperties);
+				break;
+			case SOURCE_ADDRESS:
+				createStream.setSourceAddress(value);
+
+				addAdvanceControlProperties(advancedControllableProperties,
+						createText(stats, streamControllingGroup + StreamControllingMetric.SOURCE_ADDRESS.getName(), value));
+				populateCancelButtonForCreateStream(stats, advancedControllableProperties);
+				break;
+			case SOURCE_PORT:
+				String sourcePort = DecoderConstant.EMPTY;
+				try {
+					Integer sourcePortIntValue = Integer.parseInt(value);
+					if (sourcePortIntValue < DecoderConstant.MIN_PORT) {
+						sourcePortIntValue = DecoderConstant.MIN_PORT;
+					}
+					if (sourcePortIntValue > DecoderConstant.MAX_PORT) {
+						sourcePortIntValue = DecoderConstant.MAX_PORT;
+					}
+					sourcePort = sourcePortIntValue.toString();
+				} catch (Exception e) {
+					if (logger.isWarnEnabled()) {
+						logger.warn("Invalid port value", e);
```

Comment:
Here, and other similar code snippets: is it possible to provide more details for an error message, plus is it ok to do that for any Exception being thrown? May there be other reasons for other exceptions here?

Reply:
I fixed it